gravitee-apim-console-webui/src/management/api/creation/v4/steps/step-2-entrypoints/step-2-entrypoints-2-config.component.ts

```diff
@@ -113,13 +112,24 @@ export class Step2Entrypoints2ConfigComponent implements OnInit, OnDestroy {
         paths = paths.map(({ path, host, overrideAccess }) => ({ path, host, overrideAccess }));
       }
 
-      const listeners: HttpListener[] = [
-        {
-          type: 'http',
-          paths,
+      // Get distinct listener types
+      const listenersType = [...new Set(previousPayload.selectedEntrypoints.map(({ supportedListenerType }) => supportedListenerType))];
+
+      const listeners: Listener[] = listenersType.reduce((listeners, listenersType) => {
```

Comment:
haha a little  🤯 
as we have only 2 types we can build them separately 🤸‍♂️ 
---
But this way it's good for me